test/APIBasedAuth.test.ts

```diff
@@ -124,10 +258,112 @@ describe('with auth successfully created', () => {
       password: 'password',
       username: 'email'
     });
+
+    expect(globals.window.localStorage.setItem).toHaveBeenCalledTimes(3);
+    expect(globals.window.localStorage.setItem).toHaveBeenNthCalledWith(
+      1,
+      DEFAULT_ACCESS_TOKEN_KEY,
+      mockToken.accessToken
+    );
+    expect(globals.window.localStorage.setItem).toHaveBeenNthCalledWith(
+      2,
+      DEFAULT_ID_TOKEN_KEY,
+      mockToken.idToken
+    );
+    expect(globals.window.localStorage.setItem).toHaveBeenNthCalledWith(
+      3,
+      DEFAULT_REFRESH_TOKEN_KEY,
+      mockToken.refreshToken
+    );
+  });
+
+  test('initiatePasswordAuth success with promise storage', async () => {
+    globals.window.localStorage.setItem.mockImplementation(
+      async (_key: string, value: string) => {
+        await Promise.resolve(value);
+      }
+    );
+
+    await auth.initiatePasswordAuth({
+      password: 'password',
+      username: 'email'
+    });
+
+    expect(clientAxios.post).toHaveBeenCalledWith('/login', {
+      clientId: params.clientId,
+      password: 'password',
+      username: 'email'
+    });
+
+    expect(globals.window.localStorage.setItem).toHaveBeenCalledTimes(3);
+    expect(globals.window.localStorage.setItem).toHaveBeenNthCalledWith(
+      1,
+      DEFAULT_ACCESS_TOKEN_KEY,
+      mockToken.accessToken
+    );
+    expect(globals.window.localStorage.setItem).toHaveBeenNthCalledWith(
+      2,
+      DEFAULT_ID_TOKEN_KEY,
+      mockToken.idToken
+    );
+    expect(globals.window.localStorage.setItem).toHaveBeenNthCalledWith(
+      3,
+      DEFAULT_REFRESH_TOKEN_KEY,
+      mockToken.refreshToken
+    );
   });
 
-  test('logout does not crash for coverage', () => {
-    auth.logout();
+  test('logout does not crash for coverage', async () => {
```

Comment:
Can you remove the "for coverage" labels here an elsewhere? All tests are added for coverage =)